bittide-tools/clockcontrol/plot/Main.hs

```diff
@@ -152,6 +154,35 @@ data DataPoint (n :: Nat) (m :: Nat) =
       -- ^ the stability indicators for each of the elastic buffers
     }
 
+instance (KnownNat n, KnownNat m) => ToNamedRecord (DataPoint n m) where
+  toNamedRecord DataPoint{..} = HM.fromList $
+    fmap (\(x, y) -> (BSC.pack x, BSC.pack y)) $
+      [ ("Index", show dpIndex)
+      , ("Synchronized Time (fs)", show $ toInteger dpGlobalTime)
+      , ("Local Clock time (fs)", show $ toInteger dpLocalTime)
+      , ("Clock Period Drift (fs)", show $ toInteger dpDrift)
+      , ("Integrated FINC/FDECs", show dpCCChanges)
+      , ("Reframing State", rf2bs dpRfStage)
+      ] <>
+      [ ("EB " <> show i, show $ toInteger x)
+      | (i, x) <- Vec.toList $ Vec.zip Vec.indicesI dpDataCounts
+      ] <>
+      [ (show i <> " is stable", b2bs $ stable x)
+      | (i, x) <- Vec.toList $ Vec.zip Vec.indicesI dpStability
+      ] <>
+      [ (show i <> " is settled", b2bs $ settled x)
+      | (i, x) <- Vec.toList $ Vec.zip Vec.indicesI dpStability
+      ]
+   where
+    b2bs = \case
+      False -> "0"
+      True  -> "1"
+
+    rf2bs = \case
+      RSDetect -> "0"
+      RSWait   -> "1"
+      RSDone   -> "2"
```

Comment:
Why not:

```suggestion
    rf2bs = \case
      RSDetect -> "Detect"
      RSWait   -> "Wait"
      RSDone   -> "Done"
```

?

Reply:
The numbers are easier to parse and don't distract too much if you look at the CSV file with an editor, but I can also produce strings instead. Any preference with respect to that?

Reply:
I think I prefer strings, just because you don't need a separate (mental) mapping when looking at / parsing the CSV file.